ui/src/pages/frontsite/components/Document/DocumentLinks.js

```diff
@@ -1,35 +1,80 @@
 import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
 import { Divider, List } from 'semantic-ui-react';
 import PropTypes from 'prop-types';
+import { DownloadLink } from './DownloadLink';
 
 export class DocumentLinks extends Component {
   constructor(props) {
     super(props);
-    this.links = props.links;
+    this.linkCount = 0;
   }
 
-  renderUrlsList = () => {
-    return this.links.map(link => (
-      <List.Item href={link.value}>
-        <List.Content>{link.description}</List.Content>
-      </List.Item>
-    ));
+  hasMaxLinks = () => {
+    return (
+      this.linkCount > 0 &&
+      this.props.showMaxLinks &&
+      this.linkCount >= this.props.showMaxLinks
+    );
+  };
+
+  renderReadableList = () => {
+    return this.props.dividers ? <p>None.</p> : null;
+  };
+
+  renderDownloadableList = () => {
+    return (
+      <List>
+        {this.props.eitems.hits.map(eitem =>
+          eitem.files.map(file => {
+            const hideLink = this.hasMaxLinks();
+            this.linkCount++;
+            return hideLink ? null : (
+              <List.Item key={file.file_id}>
+                <List.Icon name="download" />
+                <List.Content>
+                  Download <DownloadLink eitem={eitem} filename={file.key} />
+                </List.Content>
+              </List.Item>
+            );
+          })
+        )}
+      </List>
+    );
+  };
+
+  renderShowAll = () => {
+    return (
+      <Link to={{}} onClick={this.props.onShowAll}>
```

Comment:
Does it have to be a `Link`? If I understand correctly this is the href that takes you to the relevant pane with all the download links, an `a` could be used since its not used for routing.

Reply:
You could probably make it work with `a` too but then you'd need to stop the event propagation, no? `Link` was more convenient in this case.